src/behavior.ts

```diff
@@ -127,23 +127,33 @@ export class Behavior {
         return selectedIds.some((value: ISelectionId) => value.includes(<ISelectionId>dataPoint.identity));
     }
 
-    private renderDataPointSelection(): void {
+    private renderDataPoints(): void {
         const { arcSelection, chordSelection } = this.options;
 
         chordSelection.style("opacity", Behavior.DimmedOpacity);
 
         arcSelection.style("opacity", (arcDescriptor: ChordArcDescriptor, arcIndex: number) => {
             const isArcSelected = arcDescriptor.selected;
+            const chords: Chords = <Chords>chordSelection.data();
 
             chordSelection
-                .filter((chordLink: any) => {
+                .filter((chordLink: Chord) => {
+                    const hasHighlights = this.options.highlightsMatrix[chordLink.source.index][chordLink.target.index] > 0;
+                    if (hasHighlights) return true;
+
```

Comment:
what it arc selection and what is chord selection?
Also, this doesn't seem right
return (chordLink.source.index === arcIndex && isArcSelected
                        || chordLink.target.index === arcIndex)
                        && isArcSelected;

Reply:
refactored